### ISSUE TITLE
Missing definition files in `dist` folder

### ISSUE THREAD

**Issue by frank-weindel** [qmhc/vite-plugin-dts#184]
### Describe the bug

I have some manual `d.ts` definition files that import some other manual `d.ts` files that do not have corresponding `.js` code files. Specifically in my reproduction below I have these files in my `src` directory:

```
src
├── types   <------ NONE OF THE FILES IN THIS DIRECTORY ARE COPIED TO `dist`
│   ├── README.md
│   ├── lng.components.namespace.d.ts
│   ├── lng.d.ts
│   ├── lng.shaders.c2d.namespace.d.ts
│   ├── lng.shaders.namespace.d.ts
│   ├── lng.textures.namespace.d.ts
│   ├── lng.tools.namespace.d.ts
│   └── lng.types.namespace.d.ts
├── internalTypes.d.mts <------ ALSO MISSING
├── commonTypes.d.mts <------ ALSO MISSING
├── lightning.d.mts
├── lightning.mjs
└── ...
```

**src/lightning.d.mts**
```ts
// ...
import * as lng from "./types/lng"; // <----- This `d.ts` file and the files it depends on are not copied over to `dist`

/**
 * This seeming nonsense helps us export all the types with the same names/structure as
 * their runtime counterparts.
 */
declare namespace lng_ {
  export import lng_ = lng;
}

export default lng_.lng_;
```

**src/types/lng.d.ts**
```ts
// ...
import * as shaders from './lng.shaders.namespace';
import * as textures from './lng.textures.namespace';
import * as components from './lng.components.namespace';
import * as tools from './lng.tools.namespace';
import * as types from './lng.types.namespace';
// ^ ----- None of these are copied over either

export {
  Application,
  Component,
  Base,
  Utils,
  StageUtils,
  Element,
  Tools,
  Stage,
  ElementCore,
  ElementTexturizer,
  Texture,
  EventEmitter,
  shaders,
  textures,
  components,
  tools
};

// `types` has to be exported as type so TS/IDEs don't allow you access it from runtime context
export type {
  types
};
```

As mentioned in the comments of the code above, none of the files in the `src/types` directory are copied over to `dist`. Which causes TypeScript to throw errors because of missing dependencies in any application that uses this library as a dependency.

```
dist
├── index.d.ts <--- Definition file copied over by vite-plugin-dts
├── ...
└── src <--- Definition files copied over by vite-plugin-dts
    ├── EventEmitter.d.mts
    ├── EventEmitter.d.mts.map
    ├── animation
    ├── application
    ├── components
    ├── flex
    ├── lightning.d.mts
    ├── lightning.d.mts.map
    ├── platforms
    ├── renderer
    ├── textures
    ├── tools
    └── tree
                ^----------- Note how the `types` directory is completely missing.
                             As well as `internalTypes.d.mts`/`commonTypes.d.mts`
```

I believe in 2174868158766fd16a72f9f44b979d2c62d8ab07 (prompted by #178) is where you added the copying of manual .d.ts files.

Thanks!

### Reproduction

https://stackblitz.com/~/github.com/frank-weindel/lightning/tree/vite-plugin-dts-bug-3

### Steps to reproduce

1. npm install
2. npm run build:lightning

### System Info

```shell
System:
    OS: macOS 12.2.1
  Binaries:
    Node: 16.14.0 - ~/.nodenv/versions/16.14.0/bin/node
    npm: 8.3.1 - ~/.nodenv/versions/16.14.0/bin/npm
  npmPackages:
    @vitejs/plugin-legacy: ^3.0.1 => 3.0.1 
    vite: ^4.0.4 => 4.0.4 
    vite-plugin-dts: ^2.0.2 => 2.0.2
```

### Validations

- [X] Read the [FAQ](https://github.com/qmhc/vite-plugin-dts#faq).
- [X] Check that there isn't [already an issue](https://github.com/qmhc/vite-plugin-dts/issues) that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by qmhc** [comment]
You need to add `copyDtsFiles: true` to the plugin options.

**Comment by saurabhnemade** [comment]
I also observed similar issue. may be that one is also connected to this issue.

Here is how it looks on my end:

config on vite.config
```
  plugins: [
    react(),
    dts({
      insertTypesEntry: false
    })
  ],
```

Files generated after first run (no previous build present):
<img width="385" alt="Screenshot 2023-06-03 at 17 10 24" src="https://github.com/qmhc/vite-plugin-dts/assets/17445338/6cceafb9-cb5f-4ff5-bb1c-ea7f8e3a2224">

Files generated after second run (previous build present):

<img width="279" alt="Screenshot 2023-06-03 at 17 11 17" src="https://github.com/qmhc/vite-plugin-dts/assets/17445338/f55d07c9-654a-4607-aedc-8be4f02bc860">

**Comment by saurabhnemade** [comment]
For me downgrading to ` "vite-plugin-dts": "1.7.3",` resolved the issue.

**Comment by qmhc** [comment]
Fixed in `3.0.1`. You need to specify `copyDtsFiles: true`.
